dbcon/joblist/columncommand-jl.cpp

```diff
@@ -140,9 +188,22 @@ void ColumnCommandJL::createCommand(ByteStream& bs) const
     colType.serialize(bs);
     bs << (uint8_t) isScan;
     bs << traceFlags;
-    bs << filterString;
-    bs << BOP;
-    bs << filterCount;
+    if (isDict() && fContainsRanges)
+    {
+        // XXX: we should discern here between IS (NOT) NULL and other filters.
+        ByteStream empty;
+        auto zeroFC = filterCount;
```

Comment:
> 
> 
> Can you please clarify, why do you initialize the `zeroFC` with `filterCount` and then set it to 0?
> 
> Current code:
> 
> ```
>  movl  filterCount(%rip), %eax
>  movl  %eax, -4(%rbp)
>  movl  $0, -4(%rbp)
> ```
> 
> Should be: zeroFC = 0
> 
> ```
>  movl  $0, -4(%rbp)
>  movl  -4(%rbp), %eax
> ```

It is because of "auto".

I need zeroFC to be same physical type as filterCount and, should type of filterCount change, the code I wrote will not make author of such change search for possible differences in behavior between different column types.

Reply:
I'm not familiar with plans to have 2^32 amount of filters =)
I think this should work even for this scenario.
`bs << decltype(filterCount) (0);`